code/go/internal/validator/semantic/kibana_matching_object_ids.go

```diff
@@ -32,18 +32,26 @@ func ValidateKibanaObjectIDs(pkgRoot string) ve.ValidationErrors {
 	for _, objectFile := range objectFiles {
 		filePath := objectFile.Path()
 
-		idPath := "$.id"
-		// Special case: object is of type 'security_rule'
-		if filepath.Base(filepath.Dir(filePath)) == "security_rule" {
-			idPath = "$.rule_id"
-		}
-
-		objectID, err := objectFile.Values(idPath)
+		objectID, err := objectFile.Values("$.id")
 		if err != nil {
 			errs = append(errs, errors.Wrapf(err, "unable to get Kibana object ID in file [%s]", filePath))
 			continue
 		}
 
+		// Special case: object is of type 'security_rule'
+		if filepath.Base(filepath.Dir(filePath)) == "security_rule" {
+			ruleID, err := objectFile.Values("$.attributes.rule_id")
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "unable to get rule ID in file [%s]", filePath))
+				continue
+			}
+
+			if ruleID != objectID {
```

Comment:
There's no reason these have to match.

The `rule_id` is only relevant to the Security solution.
The top level `id` is the saved object ID. It's possible that in the future we want better named SO ids. But we can loosen up this spec up then if that happens.